sw/applications/include/layer.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <stdint.h>
+
+typedef enum { CONV2D, GEMM, POOLING, BATCH_NORM } LAYER_TYPE;
```

Comment:
Not sure if we adhere to any kind of C style guide, but i like to append any typedefs with `_t` and keep them lowercase. e.g. `layer_type_t` and `activation_t`.